src/spl/parameter_space.h

```diff
@@ -223,6 +202,23 @@ class ParameterSpace {
     }
   }
 
+  void RecreateBasisFunctions() {
+    for (int current_dim = 0; current_dim < DIM; ++current_dim) {
```

Comment:
Please rename `current_dim` to `current_dimension`. Same for following variables.